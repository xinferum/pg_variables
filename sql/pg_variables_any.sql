-- Integer variables
SELECT pgv_get('vars', 'int1', NULL::int);
SELECT pgv_get('vars', 'int1', NULL::int, false);

SELECT pgv_set('vars', 'int1', 101);
SELECT pgv_set('vars', 'int2', 102);

SELECT pgv_get('vars', 'int1', NULL::int);
SELECT pgv_get('vars', 'int2', NULL::int);
SELECT pgv_set('vars', 'int1', 103);
SELECT pgv_get('vars', 'int1', NULL::int);

SELECT pgv_get('vars', 'int3', NULL::int);
SELECT pgv_get('vars', 'int3', NULL::int, false);
SELECT pgv_exists('vars', 'int3');
SELECT pgv_exists('vars', 'int1');
SELECT pgv_exists('vars2');
SELECT pgv_exists('vars');

SELECT pgv_set('vars', 'intNULL', NULL::int);
SELECT pgv_get('vars', 'intNULL', NULL::int);

-- Text variables
SELECT pgv_set('vars', 'str1', 's101'::text);
SELECT pgv_set('vars', 'int1', 's101'::text);
SELECT pgv_set('vars', 'str1', 101);
SELECT pgv_set('vars', 'str2', 's102'::text);

SELECT pgv_get('vars', 'str1', NULL::text);
SELECT pgv_get('vars', 'str2', NULL::text);
SELECT pgv_set('vars', 'str1', 's103'::text);
SELECT pgv_get('vars', 'str1', NULL::text);

SELECT pgv_get('vars', 'str3', NULL::text);
SELECT pgv_get('vars', 'str3', NULL::text, false);
SELECT pgv_exists('vars', 'str3');
SELECT pgv_exists('vars', 'str1');
SELECT pgv_get('vars', 'int1', NULL::text);
SELECT pgv_get('vars', 'str1', NULL::int);

SELECT pgv_set('vars', 'strNULL', NULL::text);
SELECT pgv_get('vars', 'strNULL', NULL::text);

-- Numeric variables
SELECT pgv_set('vars', 'num1', 1.01::numeric);
SELECT pgv_set('vars', 'num2', 1.02::numeric);
SELECT pgv_set('vars', 'str1', 1.01::numeric);

SELECT pgv_get('vars', 'num1', NULL::numeric);
SELECT pgv_get('vars', 'num2', NULL::numeric);
SELECT pgv_set('vars', 'num1', 1.03::numeric);
SELECT pgv_get('vars', 'num1', NULL::numeric);

SELECT pgv_get('vars', 'num3', NULL::numeric);
SELECT pgv_get('vars', 'num3', NULL::numeric, false);
SELECT pgv_exists('vars', 'num3');
SELECT pgv_exists('vars', 'num1');
SELECT pgv_get('vars', 'str1', NULL::numeric);

SELECT pgv_set('vars', 'numNULL', NULL::numeric);
SELECT pgv_get('vars', 'numNULL', NULL::numeric);

SET timezone = 'Europe/Moscow';

-- Timestamp variables
SELECT pgv_set('vars', 'ts1', '2016-03-30 10:00:00'::timestamp);
SELECT pgv_set('vars', 'ts2', '2016-03-30 11:00:00'::timestamp);
SELECT pgv_set('vars', 'num1', '2016-03-30 12:00:00'::timestamp);

SELECT pgv_get('vars', 'ts1', NULL::timestamp);
SELECT pgv_get('vars', 'ts2', NULL::timestamp);
SELECT pgv_set('vars', 'ts1', '2016-03-30 12:00:00'::timestamp);
SELECT pgv_get('vars', 'ts1', NULL::timestamp);

SELECT pgv_get('vars', 'ts3', NULL::timestamp);
SELECT pgv_get('vars', 'ts3', NULL::timestamp, false);
SELECT pgv_exists('vars', 'ts3');
SELECT pgv_exists('vars', 'ts1');
SELECT pgv_get('vars', 'num1', NULL::timestamp);

SELECT pgv_set('vars', 'tsNULL', NULL::timestamp);
SELECT pgv_get('vars', 'tsNULL', NULL::timestamp);

-- TimestampTZ variables

SELECT pgv_set('vars', 'tstz1', '2016-03-30 10:00:00 GMT+01'::timestamptz);
SELECT pgv_set('vars', 'tstz2', '2016-03-30 11:00:00 GMT+02'::timestamptz);
SELECT pgv_set('vars', 'ts1', '2016-03-30 12:00:00 GMT+03'::timestamptz);

SELECT pgv_get('vars', 'tstz1', NULL::timestamptz);
SELECT pgv_get('vars', 'tstz2', NULL::timestamptz);
SELECT pgv_set('vars', 'tstz1', '2016-03-30 12:00:00 GMT+01'::timestamptz);
SELECT pgv_get('vars', 'tstz1', NULL::timestamptz);

SELECT pgv_get('vars', 'tstz3', NULL::timestamptz);
SELECT pgv_get('vars', 'tstz3', NULL::timestamptz, false);
SELECT pgv_exists('vars', 'tstz3');
SELECT pgv_exists('vars', 'tstz1');
SELECT pgv_get('vars', 'ts1', NULL::timestamptz);

SELECT pgv_set('vars', 'tstzNULL', NULL::timestamptz);
SELECT pgv_get('vars', 'tstzNULL', NULL::timestamptz);

-- Date variables
SELECT pgv_set('vars', 'd1', '2016-03-29'::date);
SELECT pgv_set('vars', 'd2', '2016-03-30'::date);
SELECT pgv_set('vars', 'tstz1', '2016-04-01'::date);

SELECT pgv_get('vars', 'd1', NULL::date);
SELECT pgv_get('vars', 'd2', NULL::date);
SELECT pgv_set('vars', 'd1', '2016-04-02'::date);
SELECT pgv_get('vars', 'd1', NULL::date);

SELECT pgv_get('vars', 'd3', NULL::date);
SELECT pgv_get('vars', 'd3', NULL::date, false);
SELECT pgv_exists('vars', 'd3');
SELECT pgv_exists('vars', 'd1');
SELECT pgv_get('vars', 'tstz1', NULL::date);

SELECT pgv_set('vars', 'dNULL', NULL::date);
SELECT pgv_get('vars', 'dNULL', NULL::date);

-- Jsonb variables
SELECT pgv_set('vars2', 'j1', '[1, 2, "foo", null]'::jsonb);
SELECT pgv_set('vars2', 'j2', '{"bar": "baz", "balance": 7.77, "active": false}'::jsonb);
SELECT pgv_set('vars', 'd1', '[1, 2, "foo", null]'::jsonb);

SELECT pgv_get('vars2', 'j1', NULL::jsonb);
SELECT pgv_get('vars2', 'j2', NULL::jsonb);
SELECT pgv_set('vars2', 'j1', '{"foo": [true, "bar"], "tags": {"a": 1, "b": null}}'::jsonb);
SELECT pgv_get('vars2', 'j1', NULL::jsonb);

SELECT pgv_get('vars2', 'j3', NULL::jsonb);
SELECT pgv_get('vars2', 'j3', NULL::jsonb, false);
SELECT pgv_exists('vars2', 'j3');
SELECT pgv_exists('vars2', 'j1');
SELECT pgv_get('vars', 'd1', NULL::jsonb);

SELECT pgv_set('vars', 'jNULL', NULL::jsonb);
SELECT pgv_get('vars', 'jNULL', NULL::jsonb);

-- Array variables
SELECT pgv_set('vars', 'arr1', '{1, 2, null}'::int[]);
SELECT pgv_set('vars', 'arr2', '{"bar", "balance", "active"}'::text[]);
SELECT pgv_set('vars2', 'j1', '{1, 2, null}'::int[]);

SELECT pgv_get('vars', 'arr1', NULL::int[]);
SELECT pgv_get('vars', 'arr2', NULL::int[]);
SELECT pgv_set('vars', 'arr1', '{"bar", "balance", "active"}'::text[]);
SELECT pgv_set('vars', 'arr1', '{3, 4, 5}'::int[]);
SELECT pgv_get('vars', 'arr1', NULL::int[]);

SELECT pgv_get('vars', 'arr3', NULL::int[]);
SELECT pgv_get('vars', 'arr3', NULL::int[], false);
SELECT pgv_exists('vars', 'arr3');
SELECT pgv_exists('vars', 'arr1');
SELECT pgv_get('vars2', 'j1', NULL::int[]);

SELECT pgv_set('vars', 'arrNULL', NULL::int[]);
SELECT pgv_get('vars', 'arrNULL', NULL::int[]);

-- Manipulate variables
SELECT * FROM pgv_list() order by package, name;

SELECT pgv_remove('vars', 'int3');
SELECT pgv_remove('vars', 'int1');
SELECT pgv_get('vars', 'int1', NULL::int);
SELECT pgv_exists('vars');

SELECT pgv_remove('vars2');
SELECT pgv_get('vars2', 'j1', NULL::jsonb);
SELECT pgv_exists('vars2');

SELECT * FROM pgv_list() order by package, name;

SELECT pgv_free();
SELECT pgv_exists('vars');

SELECT * FROM pgv_list() order by package, name;
