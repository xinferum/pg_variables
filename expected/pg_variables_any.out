-- Integer variables
SELECT pgv_get('vars', 'int1', NULL::int);
ERROR:  unrecognized package "vars"
SELECT pgv_get('vars', 'int1', NULL::int, false);
 pgv_get 
---------
        
(1 row)

SELECT pgv_set('vars', 'int1', 101);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'int2', 102);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'int1', NULL::int);
 pgv_get 
---------
     101
(1 row)

SELECT pgv_get('vars', 'int2', NULL::int);
 pgv_get 
---------
     102
(1 row)

SELECT pgv_set('vars', 'int1', 103);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'int1', NULL::int);
 pgv_get 
---------
     103
(1 row)

SELECT pgv_get('vars', 'int3', NULL::int);
ERROR:  unrecognized variable "int3"
SELECT pgv_get('vars', 'int3', NULL::int, false);
 pgv_get 
---------
        
(1 row)

SELECT pgv_exists('vars', 'int3');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars', 'int1');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_exists('vars2');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_set('vars', 'intNULL', NULL::int);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'intNULL', NULL::int);
 pgv_get 
---------
        
(1 row)

-- Text variables
SELECT pgv_set('vars', 'str1', 's101'::text);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'int1', 's101'::text);
ERROR:  variable "int1" requires "integer" value
SELECT pgv_set('vars', 'str1', 101);
ERROR:  variable "str1" requires "text" value
SELECT pgv_set('vars', 'str2', 's102'::text);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'str1', NULL::text);
 pgv_get 
---------
 s101
(1 row)

SELECT pgv_get('vars', 'str2', NULL::text);
 pgv_get 
---------
 s102
(1 row)

SELECT pgv_set('vars', 'str1', 's103'::text);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'str1', NULL::text);
 pgv_get 
---------
 s103
(1 row)

SELECT pgv_get('vars', 'str3', NULL::text);
ERROR:  unrecognized variable "str3"
SELECT pgv_get('vars', 'str3', NULL::text, false);
 pgv_get 
---------
 
(1 row)

SELECT pgv_exists('vars', 'str3');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars', 'str1');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_get('vars', 'int1', NULL::text);
ERROR:  variable "int1" requires "integer" value
SELECT pgv_get('vars', 'str1', NULL::int);
ERROR:  variable "str1" requires "text" value
SELECT pgv_set('vars', 'strNULL', NULL::text);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'strNULL', NULL::text);
 pgv_get 
---------
 
(1 row)

-- Numeric variables
SELECT pgv_set('vars', 'num1', 1.01::numeric);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'num2', 1.02::numeric);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'str1', 1.01::numeric);
ERROR:  variable "str1" requires "text" value
SELECT pgv_get('vars', 'num1', NULL::numeric);
 pgv_get 
---------
    1.01
(1 row)

SELECT pgv_get('vars', 'num2', NULL::numeric);
 pgv_get 
---------
    1.02
(1 row)

SELECT pgv_set('vars', 'num1', 1.03::numeric);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'num1', NULL::numeric);
 pgv_get 
---------
    1.03
(1 row)

SELECT pgv_get('vars', 'num3', NULL::numeric);
ERROR:  unrecognized variable "num3"
SELECT pgv_get('vars', 'num3', NULL::numeric, false);
 pgv_get 
---------
        
(1 row)

SELECT pgv_exists('vars', 'num3');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars', 'num1');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_get('vars', 'str1', NULL::numeric);
ERROR:  variable "str1" requires "text" value
SELECT pgv_set('vars', 'numNULL', NULL::numeric);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'numNULL', NULL::numeric);
 pgv_get 
---------
        
(1 row)

SET timezone = 'Europe/Moscow';
-- Timestamp variables
SELECT pgv_set('vars', 'ts1', '2016-03-30 10:00:00'::timestamp);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'ts2', '2016-03-30 11:00:00'::timestamp);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'num1', '2016-03-30 12:00:00'::timestamp);
ERROR:  variable "num1" requires "numeric" value
SELECT pgv_get('vars', 'ts1', NULL::timestamp);
         pgv_get          
--------------------------
 Wed Mar 30 10:00:00 2016
(1 row)

SELECT pgv_get('vars', 'ts2', NULL::timestamp);
         pgv_get          
--------------------------
 Wed Mar 30 11:00:00 2016
(1 row)

SELECT pgv_set('vars', 'ts1', '2016-03-30 12:00:00'::timestamp);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'ts1', NULL::timestamp);
         pgv_get          
--------------------------
 Wed Mar 30 12:00:00 2016
(1 row)

SELECT pgv_get('vars', 'ts3', NULL::timestamp);
ERROR:  unrecognized variable "ts3"
SELECT pgv_get('vars', 'ts3', NULL::timestamp, false);
 pgv_get 
---------
 
(1 row)

SELECT pgv_exists('vars', 'ts3');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars', 'ts1');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_get('vars', 'num1', NULL::timestamp);
ERROR:  variable "num1" requires "numeric" value
SELECT pgv_set('vars', 'tsNULL', NULL::timestamp);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'tsNULL', NULL::timestamp);
 pgv_get 
---------
 
(1 row)

-- TimestampTZ variables
SELECT pgv_set('vars', 'tstz1', '2016-03-30 10:00:00 GMT+01'::timestamptz);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'tstz2', '2016-03-30 11:00:00 GMT+02'::timestamptz);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'ts1', '2016-03-30 12:00:00 GMT+03'::timestamptz);
ERROR:  variable "ts1" requires "timestamp without time zone" value
SELECT pgv_get('vars', 'tstz1', NULL::timestamptz);
           pgv_get            
------------------------------
 Wed Mar 30 14:00:00 2016 MSK
(1 row)

SELECT pgv_get('vars', 'tstz2', NULL::timestamptz);
           pgv_get            
------------------------------
 Wed Mar 30 16:00:00 2016 MSK
(1 row)

SELECT pgv_set('vars', 'tstz1', '2016-03-30 12:00:00 GMT+01'::timestamptz);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'tstz1', NULL::timestamptz);
           pgv_get            
------------------------------
 Wed Mar 30 16:00:00 2016 MSK
(1 row)

SELECT pgv_get('vars', 'tstz3', NULL::timestamptz);
ERROR:  unrecognized variable "tstz3"
SELECT pgv_get('vars', 'tstz3', NULL::timestamptz, false);
 pgv_get 
---------
 
(1 row)

SELECT pgv_exists('vars', 'tstz3');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars', 'tstz1');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_get('vars', 'ts1', NULL::timestamptz);
ERROR:  variable "ts1" requires "timestamp without time zone" value
SELECT pgv_set('vars', 'tstzNULL', NULL::timestamptz);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'tstzNULL', NULL::timestamptz);
 pgv_get 
---------
 
(1 row)

-- Date variables
SELECT pgv_set('vars', 'd1', '2016-03-29'::date);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'd2', '2016-03-30'::date);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'tstz1', '2016-04-01'::date);
ERROR:  variable "tstz1" requires "timestamp with time zone" value
SELECT pgv_get('vars', 'd1', NULL::date);
  pgv_get   
------------
 03-29-2016
(1 row)

SELECT pgv_get('vars', 'd2', NULL::date);
  pgv_get   
------------
 03-30-2016
(1 row)

SELECT pgv_set('vars', 'd1', '2016-04-02'::date);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'd1', NULL::date);
  pgv_get   
------------
 04-02-2016
(1 row)

SELECT pgv_get('vars', 'd3', NULL::date);
ERROR:  unrecognized variable "d3"
SELECT pgv_get('vars', 'd3', NULL::date, false);
 pgv_get 
---------
 
(1 row)

SELECT pgv_exists('vars', 'd3');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars', 'd1');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_get('vars', 'tstz1', NULL::date);
ERROR:  variable "tstz1" requires "timestamp with time zone" value
SELECT pgv_set('vars', 'dNULL', NULL::date);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'dNULL', NULL::date);
 pgv_get 
---------
 
(1 row)

-- Jsonb variables
SELECT pgv_set('vars2', 'j1', '[1, 2, "foo", null]'::jsonb);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars2', 'j2', '{"bar": "baz", "balance": 7.77, "active": false}'::jsonb);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'd1', '[1, 2, "foo", null]'::jsonb);
ERROR:  variable "d1" requires "date" value
SELECT pgv_get('vars2', 'j1', NULL::jsonb);
       pgv_get       
---------------------
 [1, 2, "foo", null]
(1 row)

SELECT pgv_get('vars2', 'j2', NULL::jsonb);
                     pgv_get                      
--------------------------------------------------
 {"bar": "baz", "active": false, "balance": 7.77}
(1 row)

SELECT pgv_set('vars2', 'j1', '{"foo": [true, "bar"], "tags": {"a": 1, "b": null}}'::jsonb);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars2', 'j1', NULL::jsonb);
                       pgv_get                       
-----------------------------------------------------
 {"foo": [true, "bar"], "tags": {"a": 1, "b": null}}
(1 row)

SELECT pgv_get('vars2', 'j3', NULL::jsonb);
ERROR:  unrecognized variable "j3"
SELECT pgv_get('vars2', 'j3', NULL::jsonb, false);
 pgv_get 
---------
 
(1 row)

SELECT pgv_exists('vars2', 'j3');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars2', 'j1');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_get('vars', 'd1', NULL::jsonb);
ERROR:  variable "d1" requires "date" value
SELECT pgv_set('vars', 'jNULL', NULL::jsonb);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'jNULL', NULL::jsonb);
 pgv_get 
---------
 
(1 row)

-- Array variables
SELECT pgv_set('vars', 'arr1', '{1, 2, null}'::int[]);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars', 'arr2', '{"bar", "balance", "active"}'::text[]);
 pgv_set 
---------
 
(1 row)

SELECT pgv_set('vars2', 'j1', '{1, 2, null}'::int[]);
ERROR:  variable "j1" requires "jsonb" value
SELECT pgv_get('vars', 'arr1', NULL::int[]);
  pgv_get   
------------
 {1,2,NULL}
(1 row)

SELECT pgv_get('vars', 'arr2', NULL::int[]);
ERROR:  variable "arr2" requires "text[]" value
SELECT pgv_set('vars', 'arr1', '{"bar", "balance", "active"}'::text[]);
ERROR:  variable "arr1" requires "integer[]" value
SELECT pgv_set('vars', 'arr1', '{3, 4, 5}'::int[]);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'arr1', NULL::int[]);
 pgv_get 
---------
 {3,4,5}
(1 row)

SELECT pgv_get('vars', 'arr3', NULL::int[]);
ERROR:  unrecognized variable "arr3"
SELECT pgv_get('vars', 'arr3', NULL::int[], false);
 pgv_get 
---------
 
(1 row)

SELECT pgv_exists('vars', 'arr3');
 pgv_exists 
------------
 f
(1 row)

SELECT pgv_exists('vars', 'arr1');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_get('vars2', 'j1', NULL::int[]);
ERROR:  variable "j1" requires "jsonb" value
SELECT pgv_set('vars', 'arrNULL', NULL::int[]);
 pgv_set 
---------
 
(1 row)

SELECT pgv_get('vars', 'arrNULL', NULL::int[]);
 pgv_get 
---------
 
(1 row)

-- Manipulate variables
SELECT * FROM pgv_list() order by package, name;
 package |   name   | is_transactional 
---------+----------+------------------
 vars    | arr1     | f
 vars    | arr2     | f
 vars    | arrNULL  | f
 vars    | d1       | f
 vars    | d2       | f
 vars    | dNULL    | f
 vars    | int1     | f
 vars    | int2     | f
 vars    | intNULL  | f
 vars    | jNULL    | f
 vars    | num1     | f
 vars    | num2     | f
 vars    | numNULL  | f
 vars    | str1     | f
 vars    | str2     | f
 vars    | strNULL  | f
 vars    | ts1      | f
 vars    | ts2      | f
 vars    | tsNULL   | f
 vars    | tstz1    | f
 vars    | tstz2    | f
 vars    | tstzNULL | f
 vars2   | j1       | f
 vars2   | j2       | f
(24 rows)

SELECT pgv_remove('vars', 'int3');
ERROR:  unrecognized variable "int3"
SELECT pgv_remove('vars', 'int1');
 pgv_remove 
------------
 
(1 row)

SELECT pgv_get('vars', 'int1', NULL::int);
ERROR:  unrecognized variable "int1"
SELECT pgv_exists('vars');
 pgv_exists 
------------
 t
(1 row)

SELECT pgv_remove('vars2');
 pgv_remove 
------------
 
(1 row)

SELECT pgv_get('vars2', 'j1', NULL::jsonb);
ERROR:  unrecognized package "vars2"
SELECT pgv_exists('vars2');
 pgv_exists 
------------
 f
(1 row)

SELECT * FROM pgv_list() order by package, name;
 package |   name   | is_transactional 
---------+----------+------------------
 vars    | arr1     | f
 vars    | arr2     | f
 vars    | arrNULL  | f
 vars    | d1       | f
 vars    | d2       | f
 vars    | dNULL    | f
 vars    | int2     | f
 vars    | intNULL  | f
 vars    | jNULL    | f
 vars    | num1     | f
 vars    | num2     | f
 vars    | numNULL  | f
 vars    | str1     | f
 vars    | str2     | f
 vars    | strNULL  | f
 vars    | ts1      | f
 vars    | ts2      | f
 vars    | tsNULL   | f
 vars    | tstz1    | f
 vars    | tstz2    | f
 vars    | tstzNULL | f
(21 rows)

SELECT pgv_free();
 pgv_free 
----------
 
(1 row)

SELECT pgv_exists('vars');
 pgv_exists 
------------
 f
(1 row)

SELECT * FROM pgv_list() order by package, name;
 package | name | is_transactional 
---------+------+------------------
(0 rows)

